File: App016_ContactManager2/app/src/main/java/com/example/app016_contactmanager2/db/AppDatabase.java
package com.example.app016_contactmanager2.db;

import androidx.room.RoomDatabase;

import com.example.app016_contactmanager2.db.entities.Contact;

@androidx.room.Database(version = 1, entities = {Contact.class})
public abstract class AppDatabase extends RoomDatabase {
    public abstract ContactDAO getContactDao();
}

File: App037_Kt_ContactManager/app/src/main/java/com/kes/app037_kt_contactmanager/room/AppDatabase.kt
package com.kes.app037_kt_contactmanager.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Contact::class], version = 1)
abstract class AppDatabase: RoomDatabase() {

    abstract val contactDAO: ContactDAO

    // Singleton
    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            synchronized(this) {
                if (INSTANCE == null) {
                    // Create DB Instance
                    INSTANCE = Room.databaseBuilder(
                        context.applicationContext,
                        AppDatabase::class.java,
                        "app_db"
                    ).build()
                }

                return INSTANCE as AppDatabase
            }
        }
    }

}